src/NUnitFramework/framework/Attributes/PlatformAttribute.cs

```diff
@@ -56,21 +56,27 @@ public class PlatformAttribute : IncludeExcludeAttribute, IApplyToTest
         /// <param name="test">The test to modify</param>
         public void ApplyToTest(Test test)
         {
+            if (!IsPlatformSupportedAndValid(test) &&
+                test.RunState != RunState.NotRunnable &&
+                test.RunState != RunState.Ignored)
```

Comment:
The test of RunState has to come first. The Runstate may have been set to NotRunnable or Ignored by other attribute that executed before this one. In that case, we don't want to check the platform at all.

I see why you did it that way, since this attribute may itself set the test to non-runnable. But  that's the case with almost every attribute. I would keep the try / catch in this method and eliminate the new method so that everything is visible in one place - something like this...

```c
if (test.RunState != RunState.NotRunnable &&
    test.RunState != Runstate.Ignored)
    {
        try
        {
            if (!platformHelper.IsPlatformSupported(test))
            {
                test.Runstate = Runstate.Skipped;
                test.Properties.Add(PropertyNames.SkipReason, platformHelper.Reason);
            }
        }
        catch(InvalidPlatformException ex)
        {
            test.RunState = Runstate.NotRunnable;
            test.Properties.Add(PropertyNames.SkipReason, ex.Message);
        }
    }
```

This example (forum code, not tested) just rearranges what you have already so that there is both a pre-check and a post-check for non-runnability.